packages/web3-shared/base/src/specs/index.ts

```diff
@@ -1134,6 +1133,8 @@ export interface ProviderState<ChainId, ProviderType, NetworkType> extends Start
     ) => Promise<Account<ChainId>>
     /** Disconnect with the provider. */
     disconnect: (providerType: ProviderType) => Promise<void>
+    /** Sign a message with persona (w or w/o popups) */
+    signWithPersona(type: SignType, message: unknown, identifier?: ECKeyIdentifier, silent?: boolean): Promise<string>
```

Comment:
can you clearify?

Reply:
The `signWithPersona` does not relate to ProviderState. Better not to add it as a public method of the state.

Reply:
I cannot find a better place to add it, because I want to avoid context exposed as a global variable that can be used everywhere. I'll add a comment to note that this is not the best place to put it.